pg/src/main/java/org/bouncycastle/openpgp/PGPSecretKeyRing.java

```diff
@@ -190,7 +190,7 @@ public PGPPublicKey getPublicKey(
 
         for (int i = 0; i != extraPubKeys.size(); i++)
         {
-            PGPPublicKey    k = (PGPPublicKey)keys.get(i);
```

Comment:
Here I guess the code accidentally accessed the wrong list.

Reply:
Thanks. Applied.